### DISCUSSION TITLE
Set CORS in my Vercel app

### DISCUSSION THREAD

**Discussion by iamjonasmarlo** [vercel/community#1550] [Help] [answered]
### Summary

I am trying to send a post request to a subdomain using Axion (in a React Create app application). Unfortunately, I always get a CORS error:

_Access to XMLHttpRequest at 'https://api.mysite.com' from origin 'https://www.mysite.com' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource._

```
axios.post('https://mysite.com', {
    data
}, {
    headers: {
        'Content-Type': 'multipart/form-data',
    },
    withCredentials: false,
    crossdomain: true
})
.then(function(response) {
    console.log(response)
    setResponse(true);
});
```

I tried to configure the header using the vercel.json file:

```
{
  "headers": [
    {
      "source": "/",
      "headers": [
        { "key": "Access-Control-Allow-Credentials", "value": "true" },
        { "key": "Access-Control-Allow-Origin", "value": "*" },
        { "key": "Access-Control-Allow-Methods", "value": "GET,OPTIONS,PATCH,DELETE,POST,PUT" },
        { "key": "Access-Control-Allow-Headers", "value": "X-CSRF-Token, X-Requested-With, Accept, Accept-Version, Content-Length, Content-MD5, Content-Type, Date, X-Api-Version" }
      ]
    }
  ]
}
```

Unfortunately, this did not help. How do I allow my application to send a mail request to a subdomain?

**Comment by amyegan** [comment, marked as answer]
Hi Jonas. It looks like your current `headers` config will only apply the headers to the root and not any sub-paths. Try changing the source to `/api/(.*)` or [adding CORS headers in the Serverless Function](https://vercel.com/guides/how-to-enable-cors#enabling-cors-in-a-single-node.js-serverless-function) directly.
